README.md

```diff
@@ -531,26 +532,74 @@ Worker Groups, the default worker node groups for this module, are based on an o
-Using an ASG for each AZ bypasses the autoscaling issues in AWS.
+All of these issues can be resolved by using Worker Group Launch Templates instead, configured with a template for mixed types of on-demand and spot workers.
 Furthermore, we are also able to specify several types of machines that are suitable for spot instances rather than just one.
 Using only one often results in Spot instances not being able to be provisioned, and this greatly reduces the occurence of this happening, as well as allowing for lower spot prices.
 
 #### Enabling Worker Group Launch Templates
 
 To use the Worker Group Launch Template, set the variable `enable_worker_groups_launch_template` to `true`, and define an array of instance types allowed.
 
-When using autoscaling with Launch Templates per AZ, the min and max number of nodes is per zone.
-These values can be adjusted by using the variables `lt_desired_nodes_per_subnet`, `lt_min_nodes_per_subnet`, and `lt_max_nodes_per_subnet`
+These values can be adjusted by using the variable `workers`
+User can set k8s labels and taints (merges and passes via `kubelet_extra_args`) for different purposes.
+
+```
+workers_template_defaults = {
+    override_instance_types = var.allowed_spot_instance_types
+    root_encrypted          = var.encrypt_volume_self
+    instance_type           = var.node_machine_type
+    autoscaling_enabled     = "true"
+    public_ip               = true
+    spot_price              = (var.enable_spot_instances ? var.spot_price : null)
+    subnets                 = (var.create_vpc ? module.vpc.public_subnets : var.subnets)
+
+    root_volume_type = var.volume_type
+    root_volume_size = var.volume_size
+    root_iops        = var.iops
+
+    on_demand_base_capacity = var.on_demand_base_capacity
+    asg_min_size            = var.min_node_count
+    asg_max_size            = var.max_node_count
+    asg_desired_capacity    = var.desired_node_count
+    kubelet_extra_args      = "--node-labels=node.kubernetes.io/lifecycle=`curl -s http://169.254.169.254/latest/meta-data/instance-life-cycle`"
+
+    tags = [
+      {
+        key                 = "k8s.io/cluster-autoscaler/enabled"
+        propagate_at_launch = "false"
+        value               = "true"
+      },
+      {
+        key                 = "k8s.io/cluster-autoscaler/${var.cluster_name}"
+        propagate_at_launch = "false"
+        value               = "true"
+      }
+    ]
+  }
+```
 
 ```terraform
 module "eks-jx" {
   source  = "jenkins-x/eks-jx/aws"
   enable_worker_groups_launch_template = true
-  allowed_spot_instance_types          = ["m5.large", "m5a.large", "m5d.large", "m5ad.large", "t3.large", "t3a.large"]
-  lt_desired_nodes_per_subnet          = 2
-  lt_min_nodes_per_subnet              = 2
-  lt_max_nodes_per_subnet              = 3
+  allowed_spot_instance_types          = ["m5.large", "m5a.large", "m5d.large", "m5ad.large", "t3.large", "t3a.large"] // for all by default
```

Comment:
So what if someone has more complex needs like running a cpu workload, GPU workload and core workload. I think instead of adding more complexity in this module, we could just use this module to install jx components to an existing cluster using `create_eks` and `create_vpc` options.
I am thinking about making a breaking change to remove all the cluster creation part from this module, and just use this to create JX components. The official eks modules (which we use underneath) are much more flexible and configurable.